lib/ArduinoNative/Arduino.cpp

```diff
@@ -113,22 +113,26 @@ static const int MAX_TIME_STEP = 10;
  */
 static SimTime* gSimTime = nullptr;
 
+/** Supported long program arguments. */
+static const struct option LONG_OPTIONS[] = {{"help", no_argument, nullptr, 0},
+                                             {"serialRxCh", required_argument, nullptr, 0},
+                                             {"serialTxCh", required_argument, nullptr, 0},
+                                             {nullptr, no_argument, nullptr, 0}}; /* Marks the end. */
+
 /** Program argument default value of the robot name. */
 static const char* PRG_ARG_ROBOT_NAME_DEFAULT = "";
 
-/** Default port used for socket communications. */
-static const char* PRG_ARG_SOCKET_SERVER_PORT_DEFAULT = "65432";
+/** Program argument default value of the flag whether the ZumoComSystem is enabled or not. */
+static bool PRG_ARG_IS_ZUMO_COM_SYSTEM_ENABLED_DEFAULT = false;
 
-/** Program argument default value of the verbose flag. */
-static bool PRG_ARG_VERBOSE_DEFAULT = false;
+/** Program argument default value of the serial rx channel. */
+static const char PRG_ARG_SERIAL_RX_CH_DEFAULT[] = "1";
 
-/** Program argument default value of the serial over socket flag. */
-static bool PRG_ARG_IS_SERIAL_OVER_SOCKET_DEFAULT = false;
+/** Program argument default value of the serial tx channel. */
+static const char PRG_ARG_SERIAL_TX_CH_DEFAULT[] = "2";

```

Comment:
Would it be easier if the user could only set the RX channel, and the TX channel would be determined by (RX_Channel + 1). That way the pair of channels always matches.
We could define that RX channels can only be uneven numbers, and TX would always be even numbers. Extra question, how does Webots handle collisions of channels, when two robots choose the same channel?

Reply:
I would not do this, because it may be a blocker in case I would like to send the data to the supervisor too on channel 0.